### DISCUSSION TITLE
Failed to start Entropy Daemon based on the HAVEGE algorithm

### DISCUSSION THREAD

**Discussion by CicadaSeventeen** [SpiralLinux/SpiralLinux-project#152] [Q&A]
These error occurs every time I boot my laptop. This error also occurs on my qemu-kvm virtual machine. This is due to `haveged.service` fails to start.
I learn this software is designed as an entropy algorithm. It is not necessary. I do not know why SpiralLinux choose to pre-install this. Because do not want `sddm` get stuck for low entropy?
I try disabling this service, everythings just good.
I try reinstalling system. I find this problem not exist out-of-box.
But how can it occurs after slightly using?

**Comment by geckolinux** [comment]
Hi there. The `haveged` daemon is indeed necessary by default. Without it there can be weird delays during bootup and login to the desktop on different systems. Also, without sufficient entropy things like random number generation and SSH key generation will not be truly random and therefore easier to hack. It's not a problem on some devices because they might have a built-in hardware entropy device, but that can not be assumed for all installations.

I have seen `haveged` fail to start, but most of the time it seems to work, and if it doesn't start it won't cause any problems:

![Screenshot from 2022-10-16 17-05-14](https://user-images.githubusercontent.com/2692138/196060586-9c9d30fc-ce8b-4cba-ac1e-f7d79f7b2360.png)

------------

![Screenshot from 2022-10-16 17-05-57](https://user-images.githubusercontent.com/2692138/196060588-edfcafbb-f7f9-4cec-99d0-814984f98878.png)
